system/lib/libc/musl/src/unistd/dup2.c

```diff
@@ -14,7 +14,7 @@ int dup2(int old, int new)
 #else
 	if (old==new) {
 #ifdef __EMSCRIPTEN__
-		r = __wasi_fd_is_valid(old) ? 0 : -1;
+		r = __wasi_fd_is_valid(old) ? 0 : -EBADF;
```

Comment:
This code was previously never compiled so this bug was latent.

Reply:
FWIW, this was introduced in PR https://github.com/emscripten-core/emscripten/pull/13740.